homeassistant/components/utility_meter/sensor.py

```diff
@@ -206,6 +219,12 @@ def async_tariff_change(self, entity, old_state, new_state):
         self._state = 0
         await self.async_update_ha_state()
 
+    async def async_calibrate(self, value):
+        """Calibrate the Utility Meter with a given value."""
+        _LOGGER.debug("Calibrate %s = %s", self._name, value)
+        self._state = Decimal(value)
+        await self.async_update_ha_state()
```

Comment:
We can use `self.async_write_ha_state` to update state in the same event loop iteration.